### ISSUE TITLE
Status messages are shown with quiet kernel command line (not on first boot but following boots)

### ISSUE THREAD

**Issue by utezduyar** [systemd/systemd#4899]
### Submission type

  - [x] Bug report

*NOTE: Do not submit anything other than bug reports or RFEs via the issue tracker!*

### systemd version the issue has been seen with

> 231

### Used distribution

> Yocto

### In case of bug report: Expected behaviour you didn't see

> We use kernel command line quiet (```# cat /proc/cmdline: init=/linuxrc quiet```) and everything seems to be fine on fist boot. But on second and following boots, strangely some of the status messages are appearing. 

It is quite strange that some messages are appearing. For example ```Started multi-user.target``` status message doesn't appear. Our default.target is streaming.target which through a service files starts multi-user.target. Does this have anything to do?

```
Loading Kernel from volume 4
Booting...

zimage at:     80680010 8097A1A4
Uncompressing Linux at load address 80008000
Now, booting the kernel...
[    0.000000] Initializing cgroup subsys cpu
[    0.000000] Initializing cgroup subsys cpuacct
[    0.000000] bootconsole [early0] enabled
[    0.000000] CPU0 revision is: 0501992c (MIPS 1004Kc)
[    0.000000] Determined physical RAM map:
[    0.000000]  memory: 10000000 @ 00000000 (usable)
[    0.000000]  memory: 04000000 @ 70000000 (usable)
/linuxrc: Using kernel exported PATH='/sbin:/usr/sbin:/bin:/usr/bin'
/linuxrc: Starting init.
[   24.084712] /usr/bin/image2d[927]: Wrong stream id, do not create image2d plugin for stream id -1

/ # 
/ # cat /proc/cmdline 
init=/linuxrc quiet rootfstype=ubifs ubi.mtd=2 root=ubi0:rootfs-11 axisboot.version="AXIS NANDBOOT 12.25.1"
/ # 
/ # 
/ # 
/ # 
/ # 
/ # 
/ # 
/ # 
/ # reboot -ff
/ # [  206.597264] reboot: Restarting system
*** HARD RESET ***
Loading Kernel from volume 4
Booting...

zimage at:     80680010 8097A1A4
Uncompressing Linux at load address 80008000
Now, booting the kernel...
[    0.000000] Initializing cgroup subsys cpu
[    0.000000] Initializing cgroup subsys cpuacct
[    0.000000] bootconsole [early0] enabled
[    0.000000] CPU0 revision is: 0501992c (MIPS 1004Kc)
[    0.000000] Determined physical RAM map:
[    0.000000]  memory: 10000000 @ 00000000 (usable)
[    0.000000]  memory: 04000000 @ 70000000 (usable)
/linuxrc: Using kernel exported PATH='/sbin:/usr/sbin:/bin:/usr/bin'
/linuxrc: Starting init.
[   15.945077] /usr/bin/image2d[618]: Wrong stream id, do not create image2d plugin for stream id -1
[  OK  ] Started Network configuration on eth0.
         Starting Multicast redirection initialization...
[  OK  ] Reached target Network.
[  OK  ] Started Tampering.
[  OK  ] Started Configuration Logger.
         Starting ALSA mixer control setup...
[  OK  ] Created slice extension.slice.
         Starting Change IP over ping...
[  OK  ] Started Motion Detection Daemon.
         Starting Bonjour (mdnsd)...
         Starting Bonjour (mDNSResponderPosix)...
         Starting Data Collection Service...
[  OK  ] Started refresh uptime.
[  OK  ] Started Bandwidth measurement daemon.
[  OK  ] Started IP address and link-state monitoring daemon.
         Starting Initial analog video out pin configuration....
         Starting Post-mortem debug archive CGI...
         Starting Privacy Masking...
[  OK  ] Started Forward Password Requests to Wall Directory Watch.
         Starting io2d...
         Starting Example Addon API service...
         Starting Web service...
         Starting FTP server...
         Starting Scheduled events daemon...
         Starting Limited web page setup for locked camera...
[  OK  ] Started Parameter upgrader.
         Starting PTZaurus Port Managers service...
[  OK  ] Started Virtual input daemon.
         Starting Time source selector...
         Starting Temperature controller...
[  OK  ] Started log rotation timer.
         Starting Positioning functions...
[  OK  ] Started system status information reboot count.
[  OK  ] Started system status information boot count.
         Starting Default event producer...
[  OK  ] Created slice extension-addon.slice.
         Starting Interrupted addon install/uninstall cleanup service...
[  OK  ] Started Axis Internet Dynamic DNS.
         Starting Remote Service Client Daemon...
         Starting UPnP application...
[  OK  ] Started Change IP over ping.
[  OK  ] Started Bonjour (mdnsd).
[  OK  ] Started Post-mortem debug archive CGI.
[  OK  ] Started Interrupted addon install/uninstall cleanup service.
         Starting Active storage manager...
[  OK  ] Started Switch to multi-user.target.
[  OK  ] Started Initial analog video out pin configuration..
[  OK  ] Started FTP server.
[  OK  ] Started Multicast redirection initialization.
[  OK  ] Reached target Streaming services.
         Starting System Logger Daemon...
[  OK  ] Started Bonjour (mDNSResponderPosix).
[  OK  ] Started Time source selector.
[  OK  ] Started Limited web page setup for locked camera.
[  OK  ] Started Example Addon API service.
[  OK  ] Started Default event producer.
[  OK  ] Created slice system-ptz\x2dport\x2dmanager.slice.
         Starting PTZ port manager on 0...
[  OK  ] Started UPnP application.
[  OK  ] Started Positioning functions.
[  OK  ] Started Privacy Masking.
[  OK  ] Started Remote Service Client Daemon.
[  OK  ] Started Data Collection Service.
[  OK  ] Started Systemstatus Collection Service.
[  OK  ] Started Temperature controller.
[  OK  ] Started Scheduled events daemon.

/ # cat /proc/cmdline 
init=/linuxrc quiet rootfstype=ubifs ubi.mtd=2 root=ubi0:rootfs-10 axisboot.version="AXIS NANDBOOT 12.25.1"
```

**Comment by poettering** [comment]
what does "systemctl show -p ShowStatus" say about the boot-time status output after boot? Have you tried turning that off explicitly using systemd.show_status= for example?

Note that it defaults to "auto", which as the man page says:

> Takes a boolean argument or the constant auto. If yes, the systemd manager (PID 1) shows terse service status updates on the console during bootup.  auto behaves like false until a service fails or there is a significant delay in boot. Defaults to yes, unless quiet is passed as kernel command line option, in which case it defaults to auto.

I am pretty sure the significant delay thing happened on your system. Consider turning it off entirely to avoid this.

**Comment by poettering** [comment]
the documentation about this is crap btw, it's really hard to find the right place this is documented. I filed a bug about this now:

https://github.com/systemd/systemd/issues/4928

**Comment by utezduyar** [comment]
I don't think you can rely on "systemctl show -p ShowStatus" due to https://github.com/systemd/systemd/issues/4503

I will try explicit yes/no setting.

**Comment by utezduyar** [comment]
I have tried setting ShowStatus=yes (with kernel command line quiet) and log output is more stable in terms of things showing up or not. 

BUT, I think there is still a problem. At some point during boot systemd decides to stop on status and seems like the order is not set. I tried it twice. 

Should I open a ticket about this or should we continue from here?

![screen shot 2016-12-21 at 1 16 12 pm](https://cloud.githubusercontent.com/assets/8062069/21389431/00b2816c-c782-11e6-9040-44c437ab4d6d.png)
![screen shot 2016-12-21 at 1 31 08 pm](https://cloud.githubusercontent.com/assets/8062069/21389434/03c1748a-c782-11e6-859c-a1654201c85d.png)
